instrumentation/servlet/servlet-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/AsyncRequestCompletionListener.java

```diff
@@ -35,7 +35,8 @@ public void onComplete(RESPONSE response) {
     if (responseHandled.compareAndSet(false, true)) {
       ServletResponseContext<RESPONSE> responseContext =
           new ServletResponseContext<>(response, null);
-      instrumenter.end(context, requestContext, responseContext, null);
+      Throwable throwable = servletHelper.getAsyncException(requestContext.request());
+      instrumenter.end(context, requestContext, responseContext, throwable);
```

Comment:
does an exception thrown from a `Runnable` executed under `AsyncContext.start(Runnable)` always lead to servlet failure?

Reply:
This is a good question. If the request has already been sent then exception can not turn it back. This behaviour is the same for regular servlets. So it is possible to have requests that were ok from http clients perspective, but failed (got an exception) from servers perspective.

Reply:
good point 👍